### ISSUE TITLE
Rounding error

### ISSUE THREAD

**Issue by windmueller** [pambsoftware/beefcake#10]
For certain combination of weights, there seems to be a rounding error.

![screenshot_2015-11-18-18-46-11](https://cloud.githubusercontent.com/assets/1096199/11451545/dd077578-95c9-11e5-9a1f-5f14ef637ea0.png)

My best guess is that you use a data type like "float" despite the fact that you do not require high precision floating point operations. See for example ["Effective Java", Item 48](https://books.google.de/books?id=ka2VUBqHiWkC&pg=PA218&lpg=PA218&redir_esc=y#v=onepage&q&f=false).

**Comment by ripern** [comment]
You are correct. It's because we're using float. What would you recommend instead in sqlite? We need 2 decimals. A simple solution now would be to round the numbers in a logical way in the UI.

**Comment by JimRorie** [comment]
You should use decimal(10,2) (or appropriate precision) to avoid issues 
assoc with floats.

------ Original Message ------

**Comment by windmueller** [comment]
Since you just need two decimals, simply multiply the value by 100 and [store integer values](http://stackoverflow.com/a/16235787).

**Comment by ripern** [comment]
Actually the data in db looks good. However, the additions of weights to calculate the totalWeight was causing the rounding error. Have fixed it there now.

Regarding decimal(10,2) it doesn't exist in Sqlite, https://www.sqlite.org/datatype3.html, and will be seen as a NUMERIC type. People seemed to be confused about how to grasp it (so am I), http://stackoverflow.com/questions/21757722/how-to-use-sqlite-decimal-precision-notation
@JimRorie Do you know if the affinity is stored somewhere in Sqlite so it actually uses that information (2 decimals) when storing a value in such a column?

Thanks for your input guys!

**Comment by JimRorie** [comment]
My sqlite is a little rusty, so I led you astray.  There doesn't appear 
to be bcd handling.  The two paths I have seen are storage of values as 
text (not a fan) and storage as integer with implied decimals (x*100 in, 
x/100 out).  I don't think explicitly casting will work.

------ Original Message ------
